Merging Shop features with Master

shoaibkhan17 opened the pull request:
Features added: 
- Spring boot endpoint authentication fix
- Added sign out feature
- Fixing routing issues
- Manually update the shops list feature
- Display user name and email in toolbar
- Store user shops on the front end side
- Remove scroll bar with DisplayShop component
- Edit shop form
- Edit ships feature + ui
- Warning message for deleting a shop
- Removing the displayShopTags class
- Add the DisplayShopTags
- Hide tags if the shop contains no tag
- Added test products to the database
- Update shop feature 
- Delete shop feature
- Discord changes feature
- Updated the UML
- Can no longer add empty tags
- Added three fonts
- Added checkout endpoint
- Worked on the backend for "CreateShop"
- Products list basic ui
- Add product, update product and delete product
- Add Shop form and request for creating shop front-end/back-end
- Added Search and Filter functionality
- Revert "Added Search and Filter functionality"
- Merging the filter feature from filterbranch
- Merge branch 'FilterBranch' into shop
- Add shop and styling fixes 
- Removed unused imports
- Toolbar redirect buttons
- Add Shop and Delete Shop bug fixes
- Signed-in user can edit only their own shops.
- Sign-in user can only edit their own products
- Shop and Cart API tests
- Add model tests for cart item

shoaibkhan17 committed:
Spring boot endpoint authentication fix

-Allow all endpoints
-Block shops/protected endpoints

shoaibkhan17 committed:
Added sign out feature

-Polished sign in and create account
-sign out feature
-Auto update and clear JWT

shoaibkhan17 committed:
Fixing routing issues

-Redirect to "/" for all links
-The react app with handle client side routing

shoaibkhan17 committed:
Manually update the shops list feature

-No need to make API calls to getShop() after deleting or adding a shop.

shoaibkhan17 committed:
Display user name and email in toolbar

-Fetch the current user from firebase.
-Display their email and name in the toolbar menu.

shoaibkhan17 committed:
Store user shops on the front end side

-Filter all the shops fetched from the backend.
-Save the user shops in another list.

shoaibkhan17 committed:
Remove scroll bar with DisplayShop compont

-Also removed the teal background on sign-in and create user pages

shoaibkhan17 committed:
Edit shop form

shoaibkhan17 committed:
Edit shop feature + ui

-Updated primary colors
-Created shop component

shoaibkhan17 committed:
Warning message for deleting a shop

-Display alert message

shoaibkhan17 committed:
Removing the diplayShopTags class

-For renaming it

shoaibkhan17 committed:
Add the DisplayShopTags

shoaibkhan17 committed:
Hide tag if the shop contains no tag

-Viability hidden

meetdigrajkar committed:
Deleting products associated to the shop

meetdigrajkar committed:
Merge pull request #20 from shoaibkhan17/firebase-auth

Updating features from firebase-auth to deleteshop branch

meetdigrajkar committed:
Added test products to fruit shop & add product, get product endpoints

shoaibkhan17 committed:
Update shop feature

-Can update the shop on the frontend side
-Returns a boolean value if the update was successful

shoaibkhan17 committed:
Delete shop and discard changes

-Delete shop frontend + backend update
-Deletes the shop from the redux store list
-Discard changes feature

shoaibkhan17 committed:
Update UML

-Removed the arrows 
-Added a parameter in the constructor

abdillahinur reviewed (commented):
Reviewed features, looks good to merge into master

shoaibkhan17 committed:
Can no longer add empty tags

shoaibkhan17 committed:
Added three fonts to the websote

-Our generation
-Rainday
-Wearedimdam

meetdigrajkar committed:
Added check out endpoint

-checkout end point (takes map of product id and quantity)
-returns new map of product ids and quantities
-

TamerIbrahim97 committed:
Worked on the backend for "CreateShop"

TamerIbrahim97 committed:
Merge remote-tracking branch 'origin/shop' into shop

shoaibkhan17 committed:
Products list basic ui

-Routes and opens the product page

abdillahinur committed:
Added Search and Filter functionality

meetdigrajkar committed:
Add product, update product and delete product

Added form for updating and adding a product to a shop. Updated the back-end endpoints for update, delete and add products.

shoaibkhan17 committed:
Bug fixes with add, delete & update product

-Wasn't rendering properly 
-Wasn't making proper requests

meetdigrajkar committed:
Add Shop form and request for creating shop front-end/back-end

meetdigrajkar committed:
Added Search and Filter functionality

merging this commit to shop branch

shoaibkhan17 committed:
Revert "Added Search and Filter functionality"

This reverts commit 16baaa3b30d7aea6c402b6ace9f3cc8b36ad2074.

shoaibkhan17 committed:
Merging the filter feature from filterbranch

shoaibkhan17 committed:
Merge branch 'FilterBranch' into shop

shoaibkhan17 committed:
Add shop and styling fixes 

-Add shop endpoint bug fix
-Display shops add button style fix
-Default shop and product images

shoaibkhan17 committed:
Removed unused imports

shoaibkhan17 committed:
Toolbar redirect buttons

-Routes back to home

meetdigrajkar committed:
Add Shop and Delete Shop bug fixes

meetdigrajkar committed:
Signed-in user can edit only their own shops.

meetdigrajkar committed:
Sign-in user can only edit their own products

meetdigrajkar committed:
Shop and Cart API tests

getShops and unauthorized create shop API test. Invalid checkout products API test.

shoaibkhan17 committed:
Add model tests for cart item

meetdigrajkar reviewed (approved):
Reviewed and verified changes
